django/cantusdb_project/main_app/templates/browse_chants.html

```diff
@@ -52,79 +52,84 @@ <h3>Browse Chants</h3>
+                                <td class="text-wrap">
+                                    <a href="{% url 'chant-detail' chant.id %}"><b>{{ chant.incipit|default:"" }}</b></a>
+                                    <p>{{ chant.manuscript_full_text_std_spelling|default:"" }}<br>
+                                    {% if chant.volpiano %}
+                                        <span style="font-family: volpiano; font-size:25px">{{ chant.volpiano|default:"" }}</span>
+                                    {% endif %}
+                                    </p>
+                                </td>
+                                <td class="text-wrap" style="text-align:center">
+                                    {% if chant.feast %}
+                                        <a href="{% url 'feast-detail' chant.feast.id %}" title="{{ chant.feast.description }}">{{ chant.feast.name|default:"" }}</a>
+                                    {% endif %}
+                                </td>
+                                <td class="text-wrap" style="text-align:center">
+                                    {% if chant.office %}
+                                        <a href="{% url 'office-detail' chant.office.id %}" title="{{ chant.office.description }}">{{ chant.office.name|default:"" }}</a>
+                                    {% endif %}
+                                </td>
                                 <td class="text-wrap" style="text-align:center">
-                                    <a href="{% url 'source-edit-chants' chant.source.id %}?pk={{ chant.id }}&folio={{ chant.folio }}&ref=chant-list">Edit</a>
+                                    {% if chant.genre %}
+                                        <a href="{% url 'genre-detail' chant.genre.id %}" title="{{ chant.genre.description }}">{{ chant.genre.name|default:"" }}</a>
+                                    {% endif %}
                                 </td>
-                            {% endif %}
-                        </tr>
-                        {% endfor %}
-                    </tbody>
-                </table>
-                {% include "pagination.html" %}
-            {% else %}
-                No chants found.
-            {% endif %}
+                                <td class="text-wrap" style="text-align:center">{{ chant.position|default:"" }}</td>
+                                <td class="text-wrap" style="text-align:center">
+                                    <a href="{{ chant.get_ci_url }}" target="_blank">{{ chant.cantus_id|default:"" }}</a>
+                                </td>
+                                <td class="text-wrap" style="text-align:center">{{ chant.mode|default:"" }}</td>
+                                <td class="text-wrap" style="text-align:center">
+                                    {% if chant.source.exists_on_cantus_ultimus %}
```

Comment:
Since we are setting this on a source-by-source basis (setting aside the discussion in the related issue about whether there will ever be chants not on CU that are in sources that do exist on CU -- I think there we've determined we just won't handle those cases at the moment), do you need to check this? Won't it be the same for every chant in the source...so you could just check from the source?

Reply:
This point also makes me realize that we might need to update the `chant_search` template, because the chant-level image links are also shown there. I realize that is not discussed in any of the related issues, so I will create a new one for a separate PR.

Reply:
This is now #1459